common/debug-interface/src/libra_trace.rs

```diff
@@ -79,30 +101,36 @@ impl Drop for TraceBlockGuard {
 #[macro_export]
 macro_rules! end_trace {
     ($stage:expr, $node:tt) => {
-        $crate::json_log::send_json_log($crate::json_log::JsonLogEntry::new(
-            $crate::libra_trace::TRACE_EVENT,
-            serde_json::json!({
-               "path": module_path!(),
-               "node": $crate::format_node!($node),
-               "stage": $stage,
-               "end": true,
-            }),
-        ));
+        let node = $crate::format_node!($node);
+        if $crate::is_selected(node.to_string()) {
```

Comment:
Looks like what we do here currently is first 'gluing' `{node, xxx}` into string and then in is_selected we are parsing it back.
I think we could do it differently - make a separate macro (similar to `format_node`) that will select first element in $node and then use it in is_selected. We can assume that first element in `$node` is always `&str`. This way even `format_node` code can be inside `if` and we can skip formatting node if it does not need to be sampled.